src/engine/sync/enginesync.h

```diff
@@ -9,12 +9,25 @@
 class InternalClock;
 class EngineChannel;
 
+#define BPM_CONFIG_GROUP "[BPM]"
+#define SYNC_LOCK_ALGORITHM_CONFIG_KEY "sync_lock_algorithm"
+
 /// EngineSync is the heart of the Mixxx Sync Lock engine.  It knows which objects
 /// (Decks, Internal Clock, etc) are participating in Sync and what their statuses
 /// are. It also orchestrates sync handoffs between different decks as they play,
 /// stop, or request their status to change.
 class EngineSync : public SyncableListener {
   public:
+    // Remove pick SyncLockAlgorithm when new beatgrid detection
+    // and editing code is committed and we no longer need the lock bpm fallback option.
+    enum SyncLockAlgorithm {
+        // New behavior, which should work if beatgrids are reliable.
+        PREFER_SOFT_LEADER,
```

Comment:
I'm not sure if I'm missing something, but isn't it irrelevant if the leader is implicit/soft or explicit? 

```suggestion
        PREFER_LEADER,
```

Reply:
The explicit leader (full crown symbol) is always preferred. This feature is about the automatically selected soft leader. 
Shall the follower follow a dynamic tempo or should both follow a steady tempo.

Reply:
Currently this preferences option is designed to be extensible for more features that will come on the table once we have rhythm detection implemented. 
Does it make sense to decouple it?